### DISCUSSION TITLE
Proposals and RFCs and PRs, Oh My!

### DISCUSSION THREAD

**Discussion by ghost** [bluesky-social/atproto#1140] [General Discussion]
Hi folks I'm writing up https://github.com/atblock-io/papers/pull/1/files at the moment to try and formalize some thoughts I was bouncing around with @bnewbold last week and get ahead of some things before federation. Since, AFAICT, this is the "first" protocol extension proposal floating around, I'd like to make sure I'm not (a) stepping on anyone's toes, and (b) running off in a silly direction with processes that don't align with your vision of how atproto ecosystem evolution will proceed.

Separately from the content of that draft RFC, specifically, I'm interested to know how you all would like to go about this, because I have... half a dozen(?) more or less concrete ideas in the works at the protocol level that I would like to start writing up formally and get feedback on. I expect others may be in this situation too, so perhaps we should make official some lightweight way of creating and publishing RFCs against atproto? Would you be interested in community contributions to stand up this process and get it rolling? Say with a new repo, or folder, or whatever makes sense to host RFCs going forward.

**Comment by ianklatzco** [comment]
A GitHub repo where RFCs are PR'd, then rendered on a website that tastes like [The Good One](https://www.rfc-editor.org/rfc/rfc1035), would be cool.

**Comment by dholms** [comment]
We wanna do this so bad. But right now we're super bandwidth-constrained & just do not have time to handle the in-bound. We also need to update & publish our specs before accepting RFCs. This is not _that_ far down the road - we just need to get our ducks in a row.

In the meantime, write down your ideas for safe keeping, feel free to chime in on discussions in the repo & when we're ready, we'll create a new repo for RFCs & proposals

**Comment by ghost** [comment]
You got it, let us know if there's anything we can do to help accelerate this, or any further leg work we can do up front.

**Comment by bnewbold** [comment]
As a follow-up on this, what we are leaning towards right now is to have long-form discussion here in the github "discussions" section of the atproto repo.

This is an appropriate place both to propose changes or extensions to atproto, or to discuss new lexicons/applications and get feedback on schema design.
